lib/ui/widgets/ata_map.dart

```diff
@@ -0,0 +1,185 @@
+
+  @override
+  State<ATAMap> createState() => ATAMapState(
+      currentMarkedLat: markedLat,
+      currentMarkedLng: markedLng,
+      centerMapLat: centerMapLat,
+      centerMapLng: centerMapLng,
+      isMoveableMarker: isMoveableMarker,
+      deviationRadius: deviationRadius,
+      titleMarkedPosition: titleMarkedPosition);
+}
+
+class ATAMapState extends State<ATAMap> {
+  CameraPosition defaultCamera;
+  static const double DEFAULT_ZOOM = 17;
+  static const int TIMEOUT_PIN_MARKER_MAP_LOADED = 4;
+  Completer<GoogleMapController> _controller = Completer();
+
+  Set<Marker> _markers = Set();
+  Set<Circle> _circles = Set();
+  final double centerMapLat;
+  final double centerMapLng;
+  double currentLocationLat;
+  double currentLocationLng;
+  double currentMarkedLat;
+  double currentMarkedLng;
+  double deviationRadius;
+
+  BitmapDescriptor markedLocationIcon;
+  final bool isMoveableMarker;
+  final String titleMarkedPosition;
+
+  ATAMapState({this.currentMarkedLat, this.currentMarkedLng, this.centerMapLat, this.centerMapLng, this.isMoveableMarker, this.deviationRadius, this.titleMarkedPosition}) {
+    defaultCamera = CameraPosition(
+      target: LatLng(centerMapLat, centerMapLng),
+      zoom: DEFAULT_ZOOM,
+    );
+  }
+  @override
+  void initState() {
+    super.initState();
+    _setCustomMapIcons();
+  }
+
+  void _setCustomMapIcons() async {
+    markedLocationIcon = await BitmapDescriptor.fromAssetImage(ImageConfiguration(size: Size(128, 128)), 'assets/images/marked-location-icon.png');
+  }
+
+  void _goToCurrentLocation() {
+    setState(() async {
+      try {
+        var currentPosition = await Geolocator().getCurrentPosition();
+        final GoogleMapController controller = await _controller.future;
+        controller.animateCamera(
+            CameraUpdate.newCameraPosition(CameraPosition(target: LatLng(currentPosition.latitude, currentPosition.longitude), zoom: DEFAULT_ZOOM)));
+        currentLocationLat = currentPosition.latitude;
+        currentLocationLng = currentPosition.longitude;
+        _addMarker(LatLng(this.currentMarkedLat, this.currentMarkedLng));
+      } on PlatformException catch (e) {
+        if (e.code == 'PERMISSION_DENIED') {
+          showDialog(
+              context: context,
+              builder: (BuildContext context) {
+                return AlertDialog(title: Text("PERMISSION_DENIED"), content: Text('GPS device is OFF, please turn ON !'));
+              });
+        }
+      }
+    });
+  }
+
+  void _addMarker(LatLng point) {
+    setState(() {
+      _markers.clear();
+      _circles.clear();
+      currentMarkedLat = point.latitude;
+      currentMarkedLng = point.longitude;
+      _markers.add(Marker(
+          markerId: MarkerId(point.toString()),
+          position: point,
+          infoWindow: InfoWindow(
+              title: isMoveableMarker == true ? '$currentMarkedLat, $currentMarkedLng' : '$titleMarkedPosition',

```

Comment:
isMoveableMarker? '$currentMarkedLat, $currentMarkedLng' : '$titleMarkedPosition',
why need to check isMoveableMarker == true, isMoveableMarker is already a bool ?